Az475 integrate lager

rzezeski opened the pull request.

Vagabond committed:
Add lager transform to compile options

rzezeski committed:
Merge branch 'master' into az475-integrate-lager

rzezeski committed:
Convert from error_logger to lager

az475

Vagabond commented on src/riak_solr_error.erl:
```diff
@@ -3,10 +3,12 @@
+    ErrMsg = lists:flatten(
+               io_lib:format("Request references undefined field: ~p~n",
+                             [FieldName])),
+    lager:error(ErrMsg),
```
I don't really like the logging being done here instead of where the error happens. Maybe this function should return the webmachine request AND the message to log?

rzezeski committed:
Get rid of whacky riak_solr_error module

az475

Vagabond committed:
Don't use lager in tests, lager is probably not running

Vagabond committed:
riak_search_schema_parser:from_eterm only has one return type

az475

The case clause here had a second clause that could never be reached
because the function never returned an error value, it errored instead.

Vagabond committed:
Don't pattern match on the result of file:read file

az475

Don't pattern match here, instead simply return the result and let the
calling code handle the error.

Vagabond committed:
Improve some error messages

az475

Vagabond committed:
Final log message tweak

az475

jonmeredith commented:
The riak_err dependency needs to be removed from rebar.config otherwise it gets pulled in when you build riak.
